lib/googleauth/impersonated_service_account.rb

```diff
@@ -87,7 +90,32 @@ class ImpersonatedServiceAccountCredentials
       #
       # @return [Google::Auth::ImpersonatedServiceAccountCredentials]
       def self.make_creds options = {}
-        new options
+        require "googleauth/default_credentials"
+        json_key_io, scope = options.values_at :json_key_io, :scope
```

Comment:
this new option is not compatible to `base_credentials` or `source_credentials`. Let's have an explicit check right at the start.